include/seqan3/range/views/kmer_hash.hpp

```diff
@@ -171,7 +191,17 @@ class kmer_hash_view : public std::ranges::view_interface<kmer_hash_view<urng_t>
         requires const_iterable_range<urng_t>
     //!\endcond
     {
-        return end();
+        if constexpr (std::ranges::common_range<urng_t const>)
+        {
+            return shape_iterator<urng_t const>{std::next(std::ranges::begin(urange),
+                                                          std::ranges::distance(urange) - shape_.size() + 1),
+                                                std::ranges::end(urange),
+                                                shape_};
+        }
+        else
+        {
+            return end();
+        }
```

Comment:
This shouldn't be necessary since `end()` already has the if constexpr?

Reply:
I can already remote the cbegin and cend in this PR.